### PR TITLE
Fix common options in blockm, surface, histogram, rose, and grdtrack

### LINKED ISSUE #1445: ESWN: Add common options to PyGMT functions and methods
**Description of the desired feature**

There are some common options from GMT that are missing from the corresponding PyGMT functions and methods. This issue tracks adding the common options to the following methods:

- [x] [pygmt.Figure.contour](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/contour.py)  @michaelgrund (#1446) 
     - U - timestamp
     - b - binary
     - d - nodata
     - e - find
     - f - coltypes
     - h - header
- [x] [pygmt.Figure.grdview](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/grdview.py) @arleaman (#1450)
     - n - interpolation
- [x] [pygmt.Figure.histogram](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/histogram.py) @beckysalvage (#1451)
     - b - binary
     - d - nodata
     - e - find
     - g - gap
     - h - header
     - w - wrap
- [x] [pygmt.Figure.image](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/image.py) @daroari (#1449)
     - U - timestamp
- [x] [pymgt.Figure.legend](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/legend.py) @obaney (#1453)
     - U - timestamp
- [x] [pygmt.Figure.plot](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/plot.py) @michaelgrund (#1502) 
     - b - binary
     - d - nodata
     - e - find
     - g - gap
     - h - header
     - w - wrap
- [x] [pygmt.Figure.plot3d](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/plot3d.py) @michaelgrund (#1503) 
     - b - binary
     - d - nodata
     - e - find
     - f - coltypes
     - g - gap
     - h - header
     - w - wrap
- [x] [pygmt.Figure.rose](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/rose.py) @srijac (#1452)
     - b - binary
     - d - nodata
     - e - find
     - g - gap
     - h - header
     - w - wrap
- [x] [pygmt.Figure.text](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/text.py) @meghanrjones (#1448)
     - U - timestamp
     - a - aspatial
     - e - find
     - h - header
     - w - wrap
- [x] [pygmt.Figure.velo](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/velo.py) @michaelgrund (#1497 ) 
     - d - nodata
     - e - find
     - h - header
     - i - incols
- [x] [pymgt.Figure.wiggle](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/wiggle.py) @michaelgrund (#1498) 
     - f - coltypes
     - t - transparency
     - w - wrap
- [x] [pygmt.grdtrack](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/grdtrack.py) @michaelgrund (#1499) 
    - b - binary
    - d - nodata
    - e - find
    - g - gap
    - h - header
    - o - outcols
    - s - skiprows
    - w - wrap
- [x] [pygmt.blockmean](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/blockm.py) and [pygmt.Figure.blockmedian](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/blockm.py) @michaelgrund (#1500) 
     - d - nodata
     - e - find
     - h - header
- [x] [pygmt.surface](https://github.com/GenericMappingTools/pygmt/blob/main/pygmt/src/surface.py) @michaelgrund (#1501)
     - b - binary
     - d - nodata
     - e - find
     - h - header
     - i - incols
     - s - skiprows

**Are you willing to help implement and maintain this feature?** Glad to help people with PRs!

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR fixes some erroneous additions from https://github.com/GenericMappingTools/pygmt/issues/1445 and prior PRs based on fixes made to GMT in https://github.com/GenericMappingTools/gmt/pull/5759. Here is the detailed list:
    - ``skiprows`` no longer in blockmean, blockmedian, blockmode, surface (not supported)
    - ``gap`` no longer in histogram, rose (not supported)
    - ``aspatial`` now documented in grdtrack
    - ``incols`` now documented in histogram
  

Closes #1445

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
I feel it's better to split the changes into two separate PRs.

File: pygmt/src/histogram.py [hunk 1]
```diff
@@ -31,8 +31,8 @@
     c="panel",
     d="nodata",
     e="find",
-    g="gap",
     h="header",
+    i="incols",
     l="label",
     p="perspective",
     t="transparency",
```
Comment (seisman) [inline pygmt/src/histogram.py:38]: Need to add `i="sequence_comma"` to `kwargs_to_strings`.

Comment (meghanrjones) [reply pygmt/src/histogram.py:38]: Thanks, fixed.

**Review by seisman — APPROVED** [review]

**Review by michaelgrund — APPROVED** [review]
Looks good!

**Pull request merged**